File: examples-sdk/python/106-pipeline-sse-standard-dialect.py
#!/usr/bin/env python3
"""106-pipeline-sse-standard-dialect — Python SDK equivalent
Compile: vil compile --from python --input 106-pipeline-sse-standard-dialect.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

pipeline = VilPipeline("IoTSensorPipeline", 3106)
pipeline.sink(port=3106, path="/stream", name="io_t_dashboard_sink")
pipeline.source(url="http://localhost:18081/api/v1/credits/stream", format="sse", name="io_t_sensor_source")
pipeline.route("sink.trigger_out", "source.trigger_in", "LoanWrite")
pipeline.route("source.sensor_data_out", "sink.sensor_data_in", "LoanWrite")
pipeline.route("source.batch_ctrl_out", "sink.batch_ctrl_in", "Copy")
pipeline.compile()

File: examples-sdk/typescript/106-pipeline-sse-standard-dialect.ts
#!/usr/bin/env tsx
// 106-pipeline-sse-standard-dialect — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 106-pipeline-sse-standard-dialect.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const p = new VilPipeline("IoTSensorPipeline", 3106);
p.sink({ port: 3106, path: "/stream", name: "io_t_dashboard_sink" });
p.source({ name: "io_t_sensor_source", url: "http://localhost:18081/api/v1/credits/stream", format: "sse" });
p.route("sink.trigger_out", "source.trigger_in", "LoanWrite");
p.route("source.sensor_data_out", "sink.sensor_data_in", "LoanWrite");
p.route("source.batch_ctrl_out", "sink.batch_ctrl_in", "Copy");
p.compile();
